packages/@lwc/compiler/src/transformers/template.ts

```diff
@@ -28,24 +28,29 @@ export default function templateTransform(
     options: NormalizedTransformOptions
 ): TransformResult {
     const {
-        experimentalDynamicComponent,
+        dynamicImportConfig,
         preserveHtmlComments,
         enableStaticContentOptimization,
         customRendererConfig,
         enableLwcSpread,
         enableScopedSlots,
+        enableDynamicComponents,
+        // TODO [#3331]: remove usage of lwc:dynamic in 246
+        experimentalDynamicDirective: deprecatedDynamicDirective,
     } = options;
-    const experimentalDynamicDirective = Boolean(experimentalDynamicComponent);
+    const experimentalDynamicDirective = deprecatedDynamicDirective ?? Boolean(dynamicImportConfig);
```

Comment:
This is actually in place to preserve existing behavior for downstream consumers.  The current behavior is to enable `lwc:dynamic` if the `experimentalDynamicComponents` option is provided.

In 244 we want to allow the LCS team to disable access to `lwc:dynamic` by exposing the `experimentalDynamicDirective` directly (they can toggle it off).

The `Boolean(experimentalDynamicComponent)` is to prevent all other downstream consumers like `lwc-test` from breaking.

Reply:
Btw I'm changing `dynamicImportConfig` back to `experimentalDynamicComponents` to prevent breaking downstream consumers and to ease the enablement of this feature.

We're planning to remove the `lwc:dynamic` directive entirely in 246 and I plan to deprecate `experimentalDynamicComponent` in favor of `dynamicImportConfig`.